File: android/src/main/java/org/ligi/passandroid/ui/edit/BarcodePickDialog.java
package org.ligi.passandroid.ui.edit;

import android.content.DialogInterface;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.View;
import org.greenrobot.eventbus.EventBus;
import org.ligi.passandroid.R;
import org.ligi.passandroid.events.PassRefreshEvent;
import org.ligi.passandroid.model.pass.BarCode;
import org.ligi.passandroid.model.pass.Pass;

public class BarcodePickDialog {

    public static void show(final AppCompatActivity context, final EventBus bus, final Pass pass, final BarCode barCode) {
        final View view = LayoutInflater.from(context).inflate(R.layout.barcode_edit, null);

        final BarcodeEditController barcodeEditController = new BarcodeEditController(view, context, barCode);

        new AlertDialog.Builder(context).setView(view)
                                        .setTitle("Edit BarCode")
                                        .setNegativeButton(android.R.string.cancel, null)
                                        .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
                                            @Override
                                            public void onClick(final DialogInterface dialog, final int which) {
                                                pass.setBarCode(barcodeEditController.getBarCode());
                                                bus.post(new PassRefreshEvent(pass));
                                            }
                                        })
                                        .show();
    }

}

File: android/src/main/java/org/ligi/passandroid/ui/edit/dialogs/BarcodePickDialog.kt
package org.ligi.passandroid.ui.edit.dialogs

import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import org.ligi.kaxt.inflate
import org.ligi.passandroid.R
import org.ligi.passandroid.model.pass.BarCode
import org.ligi.passandroid.model.pass.Pass
import org.ligi.passandroid.ui.edit.BarcodeEditController

fun showBarcodeEditDialog(context: AppCompatActivity, refreshCallback: () -> Unit, pass: Pass, barCode: BarCode) {
    val view = context.inflate(R.layout.barcode_edit)

    val barcodeEditController = BarcodeEditController(view, context, barCode)

    AlertDialog.Builder(context).setView(view)
            .setTitle(R.string.edit_barcode_dialog_title)
            .setNegativeButton(android.R.string.cancel, null)
            .setPositiveButton(android.R.string.ok) { _, _ ->
                pass.barCode = barcodeEditController.getBarCode()
                refreshCallback.invoke()
            }
            .show()
}
